### PR TITLE
ARSN-412: add support for `exists` condition for transactions

### PR DESCRIPTION
**Pull request by anurag4DSB**
**Motivation**:
We arte using the improvement for [this Vault PR](https://github.com/scality/Vault/pull/2163) and specifically [here](https://github.com/scality/Vault/pull/2163/commits/1465e88472c269ee9a94682d503e3a52c13b5ea1#diff-41150607980ec4c24c5af31dcac8f3d0dfe6df8b8e462cc9a0e802e1f4c76213R261)
We're enhancing data integrity in our IAM systems to prevent accidental deletions of mutable yet key-constrained entities. Introducing key existence checks before batch operations will help avoid conflicts and ensure data accuracy.

**Details of the Code Being Merged**:

- **Commit # 1**: Adds a new `exists` condition to transaction processing to verify key presence before operations, ensuring accuracy in entity management.

- **Commit # 2**: Implements unit tests to validate the functionality of the `exists` condition.

- **Commit # 3**: Revises the error response for the `notExists` condition from `PreConditionFailed` to `EntityAlreadyExists`, aligning with the `exists` condition which uses `NoSuchEntity`. This change will also be applied in Metadata and Vault.

- **Commit # 4**: Version bump in `package.json`.

### CODE REVIEW

**Comment by anurag4DSB** [comment]
ping

File: tests/unit/db.spec.js [hunk 1]
```diff
@@ -457,11 +469,87 @@ describe('IndexTransaction', () => {
...
-            transaction.addCondition({ exists: key1 });
+            transaction.addCondition({ like: key1 });
             done(new Error('should fail for unsupported condition, currently supported - notExists'));
         } catch (err) {
             assert.strictEqual(err.unsupportedConditionalOperation, true);
             done();
         }
     });
+
+    it('should allow batch operation if key specified in exits condition is present in db', done => {
```
Comment (BourgoisMickael) [inline tests/unit/db.spec.js]:
```suggestion
    it('should allow batch operation with key specified in exists condition is present in db', done => {
```

File: tests/unit/db.spec.js [hunk 2]
```diff
@@ -457,11 +469,87 @@ describe('IndexTransaction', () => {
...
+                next => client.get(key1, next),
+            ], (err, res) => {
+                assert.ifError(err);
+                assert.strictEqual(res[1], value2);
+                return done();
+            });
+        });
+    });
+
+    it('should not allow batch operation with key spefified in exists condition is not in db', done => {
```
Comment (BourgoisMickael) [inline tests/unit/db.spec.js]:
```suggestion
    it('should not allow batch operation with key specified in exists condition is not in db', done => {
```

**Review by BourgoisMickael — APPROVED** [review]

**Comment by anurag4DSB** [comment]
ping

File: package.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
   "engines": {
     "node": ">=16"
   },
-  "version": "7.70.29",
+  "version": "7.70.30",
```
Comment (anurag4DSB) [inline package.json]: Based on when this is ready for merge, we will be changing this tag to the latest unreleased one.

**Review by fredmnl — APPROVED** [review]

**Review by KazToozs — APPROVED** [review]

**Comment by anurag4DSB** [comment]
ping

**Pull request merged**